heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
When metrics mgr client is created, you are going to supply it a port to connect to. Supplying 0 should not be an option at all. Thus I dont understand why this method is required at all.